src/person_linkage_case_study_utils/distributed_compute.py

```diff
@@ -810,8 +810,12 @@ def to_pyarrow_large_string(df):
 
 def start_spark_cluster(
     local: bool,
+    singularity_path: str | pathlib.Path,
```

Comment:
nit: do you use anything else from pathlib? i.e. why not import as `from pathlib import Path`